src/initializers/kafka/kafka.ts

```diff
@@ -1,100 +1,134 @@
+        getLogger('orka.kafka.producer')[entry.label.toLowerCase()](message, extra);
+      },
+      ...getAuthOptions(producer)
+    });
+
+    this.producer = this.produceClient.producer(options);
     await this.producer.connect();
-    this._metadata = await this.producer.getMetadata(5000);
-    logger.info(`Kafka producer connected ${producer?.brokers?.join(', ') || brokers.join(', ')}`);
+
+    logger.info(`Kafka connected ${producer?.brokers?.join(', ')}`);
+  }
+
+  public async createConsumer({ groupId, ...rest }: KafkajsType.ConsumerConfig = {} as any) {
+    groupId ??= this.options.groupId;
+    const { brokers, clientId } = this.options;
+    this.consumeClient = new Kafka({
+      brokers,
+      clientId,
+      logCreator: (level: number) => entry => {
+        const { message, ...extra } = entry.log;
+        getLogger('orka.kafka.consumer')[entry.label.toLowerCase()](message, extra);
+      },
+      ...getAuthOptions(this.options)
+    });
+    const consumer = this.consumeClient.consumer({ groupId, ...rest });
+    await consumer.connect();
+    logger.info(`Kafka consumer(${groupId}) connected ${brokers.join(', ')}`);
+    return consumer;
+  }
+
+  public async connectAdmin() {
+    const { brokers, clientId } = this.options;
+    const kafka = new Kafka({
+      brokers,
+      clientId,
+      logCreator: (level: number) => entry => {
+        const { message, ...extra } = entry.log;
+        if (extra.error === 'Topic with this name already exists') return;
+        getLogger('orka.kafka.admin')[entry.label.toLowerCase()](message, extra);
+      },
+      ...getAuthOptions(this.options)
+    });
+    const admin = kafka.admin();
+    await admin.connect();
+    logger.info(`Kafka admin connected ${brokers.join(', ')}`);
+    return admin;
+  }
+
+  public async metadata() {
+    const admin = await this.connectAdmin();
+    const metadata = await admin.fetchTopicMetadata();
+    admin.disconnect();
+    return metadata;
   }
 
+  public async createTopics(topics: KafkajsType.ITopicConfig[]) {
+    const admin = await this.connectAdmin();
+    const responses = await Promise.all(
+      topics.map(topic =>
+        admin
+          .createTopics({ topics: [topic] })
+          .catch(e => {
+            logger.error(e);
+            return e;
+          })
+          .then(response => ({ [topic.topic]: response }))
+      )
+    );
+    await admin.disconnect();
+    logger.info(`Created topics: ${JSON.stringify(responses)}`);
+    return responses;
+  }
+
+  /**
+   * @deprecated  use producer.send instead which is native kafkajs send
+   */
   public async send(
     topic: string,
     message: string | Buffer,
-    key?: string,
+    key: string = uuid.v4(),
     partition?: number,
-    headers?: { [key: string]: string }[]
+    inputHeaders?: { [key: string]: string }[]
   ) {
-    const { key: _key, partition: _partition, offset } = await this.producer.send(
+    if (!key) key = uuid.v4();
```

Comment:
This is redundant

Reply:
does kafka.js add a key by default? Because our previous implementation did so.

Reply:
I added it because we had tests that set key as null.

Reply:
No it doesn't add it by default but it will have a value from this:
`key: string = uuid.v4(),` (line 106)

Reply:
Unless someone sets key:null as we did in the tests in order to also add headers for example.